### DISCUSSION TITLE
What's going on with 1.0.0

### DISCUSSION THREAD

**Discussion by Jeffser** [Jeffser/Alpaca#125] [Announcements]
Hi everybody, as always thanks for all the help and support, I'm very happy with how the development of Alpaca is going.

Alpaca also has been downloaded more than 10 thousand times, and it has never been downloaded less than 100 times / day.

As the first actual public app that I've develop I must say, I made something really special, and it's all thanks to you guys for your suggestions, bug reports and code contributions.

Let's get to the actual announcement 

## What's taking so long for 1.0.0?

Some of you might know this but the app is being [reviewed](https://gitlab.gnome.org/Teams/Circle/-/issues/230) for [Gnome Circle](https://circle.gnome.org/), this by itself doesn't prevent me from publishing new versions, but I want to release all the new content once (and if) the app gets approved.

## Soooo a new icon huh?

Yeah the icon has been changed officially to a HIG-compliant icon, this was done over [Gnome's Gitlab](https://gitlab.gnome.org/Teams/Circle/-/issues/230) by Tobias Bernard, many thanks to him and everybody who submitted ideas, I love the new icon.

## What's coming for 1.0.0?

Now let's get to the juicy stuff, I've been cooking a proper 1.0.0 release, this doesn't actually include everything I wanted to add, but there's some other stuff that I haven't mentioned before.

### Better model selector

The selector used to cut off options from the top bar, now it doesn't do that and it has a better format

Old|New
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/cc5ce4dd-7be6-41fa-b124-d28f76fd46c5) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/6335e61f-6122-4690-b3eb-b5b266c01572)

### Better model manager

The format for the model selector is everywhere inside the manager models dialog, I also added a description and moved some stuff

Old|New
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/102615ff-126d-45ed-ad9e-be9dbf6a5d3c) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/e3833ce6-3b28-49e6-97ee-ab2fdda86fff)

### Tag selector

Usually when the user wants to download a tag from a model a dialog pops up, now it's a different page in the same dialog, the second page also has the 'visit website' button

Old|New
:---:|:---:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/631bb9fd-53d9-4fbb-b0f9-80a8a9c90d4e) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/1b4d6d02-7c37-4f2f-b241-8a70cc491448)

### Search

So yeah, I added search for models, a lot of you have been asking me to do this and it's finally here!

Results|No results
:---:|:---:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/456da215-268e-4034-8a83-70d3dae5b2cd) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/4a2bc354-fc54-487b-96cd-4c42ff64daf3)

### New preferences dialog (feat. Bearer Token)

There's some people that self host Ollama instances with bearer tokens so I added the option for that, and also changed the preferences dialog, I merged two of the pages into just one and removed unnecessary descriptions

Old|New
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/6dd387dd-609a-4ee4-906d-5adc0854dd61) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/c6222ad2-a957-4e04-ba95-1f6487dcfefd)

### Added context menus (right click) to interact with a chat

So apparently buttons that are shown when the user hovers over an element are a bad idea because touchscreens exist, who would've thought that hahahaha

Old|New
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/776fb6cf-becd-4588-8b8d-e848f04bc4eb) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/223cc517-5f60-452d-8f64-8cbdcd0eb7c2)

### Moved around stuff on the menus

The primary menu has been moved to the left and the secondary menu now shows chat actions

Left (Secondary) (Old)|Right (Primary) (Old)|Left (Primary) (New)|Right (Secondary) (New)
:----:|:----:|:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/89acd856-b428-4c36-a09b-1b6f762afb54) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/91170ec0-b227-4a25-9630-8bf12149243d) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/b09fe1a3-24ed-4a11-b813-554100e03baf) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/76700300-2cd4-4e51-8e79-41f8b6eac1d3)

### YouTube integration

Yeah, I did it, now if you paste a YouTube video / song url that has transcriptions (not CC) Alpaca automagically adds all the transcript to the prompt in a ✨fancy way✨

Just paste the url and the prompt will show up, Alpaca detects when a video has a valid transcript but there are some videos that have some crazy custom subtitles and stuff (mostly J-POP songs as far as I've seen), those might not work

Selecting the language|Writing the prompt
:----:|:----:
![Image](https://github.com/Jeffser/Alpaca/assets/69224322/902f8ec0-4677-48c0-af42-9b9cec1aa63d) | ![Image](https://github.com/Jeffser/Alpaca/assets/69224322/c15af612-6ca1-45b6-a8d6-087264b6d30a)

### Website integration (Experimental)

Wait what? website? IS ALPACA NOT LOCAL ONLY??? Relax hahaha

What I mean by this is that there's a function that extracts the main text from a url, some websites work some others don't so your mileage may vary.

This only scraps the urls you provide, it won't visit any website you didn't tell the app to visit, and it only downloads the HTML and extracts stuff from there.

PD: yes, it saves the results as a markdown.

Prompt for confirmation|Writing the prompt
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/8291b73d-b2c7-4e63-87fc-92c152100bd6) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/30d3df20-ce12-4415-aff3-2286470cc22d)

### Title generation

Now when you send the first message in a chat it sends that prompt in a separate request to the selected model, it will generate a title based on just the prompt (and image if you send one)

I don't know how I can show this in a picture, trust me, these titles were generated.

Generated titles
:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/b78a3726-d9df-4027-91e9-b2aa61999dc7)

### Auto height for message entry

This was in part done by BurntSouls in [this PR](https://github.com/Jeffser/Alpaca/pull/112).

Basically, more text = more height

Small prompt|Big Prompt
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/ef091b71-19a1-49c9-8127-bf8e36f7f4c9) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/b843eb4f-d9ef-45d9-821d-c298c7cb5d26)

### Chat notifications

When the app isn't focused and a message is received you will get a nice notification

Notification
:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/4c819cce-ce7f-4050-98b6-ce34961d0782)

### Missing image

Since all the files and stuff is stored locally you might delete something manually, now you get this nice indicator.

Missing Image
:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/bfdc12b8-34de-498a-abe1-cb75fa77096f)

### Chat order

Now if you create a chat or you send a message it goes to the top of the list.

Trust me, New Chat **is** the newest chat
:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/1e5c8c1d-bd75-40b1-a832-88b18d29af6c)

### New file preview dialog

Now in the file preview you get this nice button to open the file in the default app, in this case since it is a YouTube transcript, it actually opens the video on your browser!

Btw yeah you can see the actual transcript from a YouTube video, neat right?

File preview dialog
:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/be3a439b-174a-458c-a1ef-239a3c48b769)

### UI tweaks

If you check out the last 50 commits you will see a lot of UI tweaks, the app has a better UI in general, you will notice once you start using it.

### Credits
New people have been added to the credits, thanks as always!

Credits
:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/f6fc3edc-f821-4a54-a6dc-696366030576)

### Stability and optimization

I removed some unused and unnecessary code, the app not only works better but actually feels much better to use

### Other stuff

I probably forgot about something so let's just say it's ✨other stuff✨ :3

----

Edit: Jul 8 2024

### Edit messages

I forgot to mentioned this in the original post but you can now edit messages! just press the button, write the new message in the box and send it using `enter` or pressing the same send button as the one used with new messages.

This won't regenerate the response but the model will take it into account on future messages.

Before Editing | Editing | After Editing
:----:|:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/b8df2668-e0d5-45cb-a5f3-8712b3c1c8a8) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/8640e9c7-d7fc-47b4-b8ca-2803a47af500) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/3c86d3bb-4325-4d7f-939d-1cc32b6f5c5d)

### Attach web images from clipboard

I also forgot to mention this, you can now paste images directly to the text view, this works only for web images (not local stuff for now), they get downloaded into a temp folder and then Alpaca attaches the files in the temp folder, the temp folder gets deleted after the user reboots their pc / phones.

Web Images
:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/91973c5c-154a-4348-bc44-2059f35f94a6)

## More new stuff!

Since the release of this post and now I've added some small stuff.

### Disclaimers

Legally speaking I'm supposed to inform the user that they are accepting the license agreement for every model they download, now, it would be really tricky to manually download every license and have them updated for every single model so instead I added these disclaimers, the user should be aware of the license of every model and I feel it is their responsibility to check that on the Ollama website.

Normal Download Page | Welcome Page
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/e51d0344-5f96-4896-8361-2dfb6c660c8a) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/25383f4d-47ea-4bd9-a8ec-a1c52b8963b9)

### File previews before sending a message

You can now preview files, websites, YouTube transcripts and images before sending them to the model, before if you clicked the file button it showed a dialog to remove the attachment, now it shows the file preview dialog with a button to delete the attachment

YouTube Transcript | Image
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/3af92b5b-d8dd-4102-ba80-4d4fd141d1d8) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/e8cab1b3-ff10-4a7f-b4ab-ab123ca7651f)

----

Edit: Jul 10 2024

## New Stuff!

The development doesn't stop!

### Better format for datetime on messages

Same Day | Same Year | Different Year
:----:|:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/744cc41a-add0-4f81-a8ad-db97971816d1) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/6718dab0-84e2-4db9-b3a2-c856ca2aebd9) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/1bfde415-5ae4-463f-9677-d4968e05e5e2)

### Error logging

Now the app logs any errors it encounters! The most common one is the `file is missing` error, the user could just delete a file from the chats directory, now it shows an error for that and other stuff. Good for debugging!

```
ERROR	[window.py | show_message] '/var/home/*****/.var/app/com.jeffser.Alpaca/data/chats/Funny fox/202406301907198778312b31903f750545539cdd930dc8373f50/pfp_pic__by_lunamoonxmoonsforeve_dfzjdqh-pre.jpg' was not found or is a directory
```

### Autohiding sidebar button!

Now when the window is wide enough to have the sidebar shown it hides the button to toggle it, if the window is narrow it comes back again, this is the same behavior of Nautilus (Gnome Files) so just open it and you will get the idea.

Wide | Narrow
:----:|:----:
![image](https://github.com/Jeffser/Alpaca/assets/69224322/89ace323-eca1-4b12-bd45-34db2a7b6048) | ![image](https://github.com/Jeffser/Alpaca/assets/69224322/9431945f-49c5-4910-9a87-b86e9899b3ab)

----

Edit: Jul 17 2024

### Logging

The logging code has been completed thanks to [imdev](https://github.com/imbev) https://github.com/Jeffser/Alpaca/pull/142

Thanks for the help

### Translations

There are a surprising amount of translations for Alpaca, I wanted to thank each one of you in this announcement!

[Alex K](https://github.com/alexkdeveloper) Russian
[Jeffser (me :3)](https://github.com/Jeffser) Spanish
[Daimar Stein](https://github.com/not-a-dev-stein) Brazilian Portuguese
[Louis Chauvet-Villaret](https://github.com/loulou64490) French
[CounterFlow64](https://github.com/CounterFlow64) Norwegian
[Aritra Saha](https://github.com/olumolu) Bengali

----

Edit: Jul 22 2024

New translation to simplified Chinese! 
Thank you [Yuehao Sui](https://github.com/8ar10der)

**Comment by Jeffser** [comment]
Oh yeah, you can now edit messages too, I forgot about it :3

**Comment by loulou64490** [comment]
That's awesome ! Big thanks!!

**Comment by olumolu** [comment]
hi i want to add some translation for this app can you help how can i do it.

**Reply by loulou64490** [reply]
Create an issue and precise which language do you want

**Reply by olumolu** [reply]
sure

**Reply by Jeffser** [reply]
Which language do you want to add?

**Reply by olumolu** [reply]
Bengali

**Reply by Jeffser** [reply]
> Bengali

Bangladesh Bengali right? I need to know the country so I can make the file with the correct language code

**Reply by olumolu** [reply]
No india but this two are same bengali just a few words are different and nothing else as the origin is same. There is no need to create bangladesh just one language if they need they can use this

**Reply by Jeffser** [reply]
https://github.com/Jeffser/Alpaca/blob/main/po/bn.po

Hi this is the file you will need to fill out to add Bengali

**Reply by Jeffser** [reply]
Also please check out this discussion
https://github.com/Jeffser/Alpaca/discussions/132

**Reply by olumolu** [reply]
Maybe it will take some time but i will try to make it fast so it can be pushed in 1.0.0

**Comment by Jeffser** [comment]
I just realized, there are like 25 changes / improvements for 1.0.0
...

I should really get some sleep

**Comment by Jeffser** [comment]
Btw I've been taking a break from development so I can get 5 projects done for uni, I will announce the release date for the update in a couple of days

**Comment by olumolu** [comment]
For better pdf support
https://github.com/opendatalab/PDF-Extract-Kit
Maybe a good option

**Comment by olumolu** [comment]
When you are planning to push the new version.
@Jeffser

**Reply by Jeffser** [reply]
Tbd

**Reply by olumolu** [reply]
Add new redme area with new app screenshots
In Screenies area

**Reply by Jeffser** [reply]
I added the first 3 new screenies to the readme

**Reply by Jeffser** [reply]
btw the new version is available on flathub

**Reply by olumolu** [reply]
Yes it is i saw that thanks for this

**Reply by olumolu** [reply]
Close this and close all 1.0.0 stuff from issue list and start a discussion on 1.1.0

**Reply by olumolu** [reply]
And let me know if anything i can help you with.

**Reply by Jeffser** [reply]
thanks just let me know if you encounter any errors

**Reply by Jeffser** [reply]
btw I will left this open in case someone wants to comment on the release of 1.0.0
